### ISSUE TITLE
revise_TSD.py: Index Error: list index out of range

### ISSUE THREAD

**Issue by cgroza** [DrosophilaGenomeEvolution/TrEMOLO#7]
Hello!

The fix about the frequency seems to have worked, thank you for the quick reply.
The sample proceeded to the TSD step.
I am now encountering this error:

```
cgroza@mallow-2 ~ % tail TSD.err
WHY params JUT-008_out/OUTSIDER/TE_DETECTION/TSD/flank_TE.fasta JUT-008_out/OUTSIDER/TE_DETECTION/TSD/sequence_TE.fasta 30 TrEMOLO.INS.11096 - -1
Feature (JAEMNL010000043.1:28051-28081) beyond the length of JAEMNL010000043.1 size (28053 bp).  Skipping.
Traceback (most recent call last):
  File "/home/cgroza/TrEMOLO/lib/TSD/revise_TSD.py", line 447, in <module>
    if not search(ID, empty_site_seq_l, empty_site_seq_r, tab_old_line, name_file_tsd, precision):
  File "/home/cgroza/TrEMOLO/lib/TSD/revise_TSD.py", line 198, in search
    positions_kr_fl = BM(kmer_r, FK_L)
  File "/home/cgroza/TrEMOLO/lib/TSD/utils.py", line 90, in BM
    pos = pos + D[1]
IndexError: list index out of range
```

I attached the TSD.err and TSD.out here.
[log.tar.gz](https://github.com/DrosophilaGenomeEvolution/TrEMOLO/files/11052185/log.tar.gz)

What could be causing this?

Any advice appreciated.
Cristian

**Comment by M-D75** [comment]
Hi,

Sorry for the delay, I had not seen the issues reported.

I have just looked at the TSD log and it seems that there are a lot of errors, when searching for the TSD sequence on the reference, there seems to be some problem with the retrieval of the arguments. I don't have enough information to know exactly what the problem is. I will change some details in the code so that the error logs are more complete and I will come back to you.

Sorry,
Mourdas

**Comment by M-D75** [comment]
update available on branch `v2.2`

**Comment by M-D75** [comment]
now available on `master`
